neqo-transport/src/connection/mod.rs

```diff
@@ -1048,10 +1062,135 @@ impl Connection {
         self.stats.borrow_mut().saved_datagrams += 1;
     }
 
+    /// Perform any processing that we might have to do on packets prior to
+    /// attempting to remove protection.
+    fn preprocess(
+        &mut self,
+        packet: &PublicPacket,
+        first: bool,
+        now: Instant,
+    ) -> Res<PreprocessResult> {
+        match (packet.packet_type(), &self.state, &self.role) {
```

Comment:
This is almost a perfect copy, except for the return codes, which are converted from the values used at the call site and the tweaks I've highlighted below.